resources/views/secretariat/evaluation-form/anonymous_questions.blade.php

```diff
@@ -1,12 +1,19 @@
 <blockquote>
     @lang('anonymous_questions.information_text')
 </blockquote>
-<form method="POST" action="{{ route('anonymous_questions.store_answer_sheet', App\Models\Semester::current()) }}">
+
+@php
+// let the current semester be found based on the periodic event itself
+// we can safely assume it is not null
+$semester = app(\App\Http\Controllers\Secretariat\SemesterEvaluationController::class)->semester();
+@endphp
+
+<form method="POST" action="{{ route('anonymous_questions.store_answers', $semester) }}">
```

Comment:
Here it is; here, we really use the periodic event to find the appropriate semester.

Reply:
Yes but completely useless (and ugly) :D . You get the semester in the blade and pass it in the argument. Then it's equivalent to just get the semester within the controller function.